wordpress-mobile/WordPress-Android
# Super Fast Media Uploads: Open Photo Picker on scan

This PR enhances #19817 by changing the launch action. Instead of launching the media library, this PR will directly launch the photo picker as if it were clicked by the user from the context menu.

- Add new NavigationAction `OpenMediaPickerForSite`
- Update `MediaBrowserActivity` to accept a new arg for launching the "open device" picker. Once the picker has been opened, the flag resets to false.
- Updates deeplink tracking logic to capture scan deep links with the proper source 

Note: This PR does not include showing an error message if siteId scanned is not valid for the user. At this moment, the user will be pushed to the MySite as happens will all other invalid deep link requests.

-----

## To Test:
#### Prerequisites: 
- Find a siteId for a site associated with a WP.com account that you will run these tests with. Jot this number down.
- Generate two test QR Codes [ I used `https://qr.io/` ]
-- Replace the "{your site id}" in the following URL with the valid site id from above. `https://apps.wordpress.com/get/?campaign=qr-code-media&data=post_id:6,site_id:{your site id}`
-- Replace the "{invalid site id}" in the following URL with an invalid site id. `https://apps.wordpress.com/get/?campaign=qr-code-media&data=post_id:6,site_id:{invalid site id}`
- Download and install JP from this PR
- Login with the account that includes the valid siteId you captured above

#### Test Media Link 
- Close or background the JP app
- Open the device camera
- Scan the QR Code with the valid site
- ✅ Verify the app launches and the media photo picker is opened (you will be prompted with permission view if you have not already allowed the proper permissions for selecting photos)
-  ✅ Verify logs contain: 🔵 Tracked: deep_linked, Properties: {"intent_action":"android.intent.action.VIEW","intent_host":"apps.wordpress.com","source":"qr-code-media","url":"apps.wordpress.com\/get"}
- Close or background the JP app
- Scan the QR Code with the invalid site
- ✅ Verify the app launches and the my site view is opened

#### Test QR Code Login still works when scanned from ME
- Open a browser on the web in incognito mode (if you are already logged in to WP.com)
- Navigate to `https://wordpress.com/log-in/` > Login via the mobile app
- STOP
- Go back to the app
- Navigate to Me > Scan Login Code
- Scan the login code shown on the web browser
- ✅ Verify the qr code login flow is launched

#### Test QR Code Login still works when scanned from camera
- Close or background the JP app
- Open a browser on the web in incognito mode (if you are already logged in to WP.com)
- Navigate to `https://wordpress.com/log-in/` > Login via the mobile app
- STOP
- Open the camera on your device
- Scan the login code shown on the web browser
- ✅ Verify the app is opened and the qr code login flow is launched
- ✅ Verify logs contain: 🔵 Tracked: deep_linked, Properties: {"intent_action":"android.intent.action.VIEW","intent_host":"apps.wordpress.com","source":"qr-code-auth","url":"apps.wordpress.com\/get"}
-----

## Regression Notes

1. Potential unintended areas of impact
Media picker doesn't launch on scan 

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing.

3. What automated tests I added (or what prevented me from doing so)
Updated `DeepLinkTrackingUtilsTest`

-----

## PR Submission Checklist:

-----

## UI Changes Testing Checklist: N/A
